### PR TITLE
fix: load gltf has no material

### PR DESCRIPTION
**Pull request by czizzy**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### Other information:

### CODE REVIEW

**Review by singlecoder — COMMENTED** [review]
+1

File: packages/loader/src/scene-loader/resources/GLTFResource.ts [hunk 1]
```diff
@@ -48,33 +48,35 @@ export class GLTFResource extends SchemaResource {
             }
           }
         };
-        for (let i = 0; i < materials.length; i++) {
-          const material = materials[i];
-          let materialResource = null;
-          let type = "";
+        if (materials && materials.length) {
```
Comment (eyworldwide) [inline packages/loader/src/scene-loader/resources/GLTFResource.ts:51]: Tip: use typescript's feature. `materials?.length`

Comment (eyworldwide) [inline packages/loader/src/scene-loader/resources/GLTFResource.ts:51]: `materials?.length` 尽量利用 ts 的语法写

Comment (czizzy) [reply packages/loader/src/scene-loader/resources/GLTFResource.ts]: fixed

**Review by zhuxudong — APPROVED** [review]
+1

**Review by gz65555 — APPROVED** [review]
+1

File: packages/loader/src/scene-loader/resources/TextureResource.ts [hunk 2]
```diff
@@ -38,9 +38,6 @@ export class TextureResource extends SchemaResource {
   setMeta() {
     if (this.resource) {
       this._meta.name = this.resource.name;
-      if (this.resource.image) {
-        this._meta.url = this.resource.image.src;
-      }
```
Comment (GuoLei1990) [inline packages/loader/src/scene-loader/resources/TextureResource.ts:43]: This resource is any, we should fix this later. @gz65555

**Review by GuoLei1990 — APPROVED** [review]
+1

**Pull request merged**